pkg/virt-api/webhooks/mutating-webhook/mutators/vmi-mutator_test.go

```diff
@@ -592,6 +592,34 @@ var _ = Describe("VirtualMachineInstance Mutator", func() {
 		Expect(vmiSpec.Domain.Memory.Guest.String()).To(Equal("4096M"))
 	})
 
+	DescribeTable("when guest-memory is set, memory-request is not set and VFIO device is requested", func(memoryOvercommit int, requestMemory string) {
+		// no limits wanted on this test, to not copy the limit to requests
+		testutils.UpdateFakeKubeVirtClusterConfig(kvInformer, &v1.KubeVirt{
+			Spec: v1.KubeVirtSpec{
+				Configuration: v1.KubeVirtConfiguration{
+					DeveloperConfiguration: &v1.DeveloperConfiguration{
+						MemoryOvercommit: memoryOvercommit,
+					},
+				},
+			},
+		})
+
+		guestMemory := resource.MustParse("3072M")
+		vmi.Spec.Domain.Memory = &v1.Memory{Guest: &guestMemory}
+		vmi.Spec.Domain.Devices.GPUs = []v1.GPU{
+			{
+				Name:       "gpu1",
+				DeviceName: "example.org/deadbeef",
+			},
+		}
+
+		_, vmiSpec, _ := getMetaSpecStatusFromAdmit(rt.GOARCH)
+		Expect(vmiSpec.Domain.Memory.Guest.String()).To(Equal("3072M"))
+		Expect(vmiSpec.Domain.Resources.Requests.Memory().String()).To(Equal(requestMemory))
```

Comment:
These quantities should be comparable to other quantities without converting to strings.